File: backend/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace backend.Models;

public class Product : BaseModel
{
  public string? Name { get; set; }
  public decimal? Price { get; set; } 
  public string? Description { get; set; } 
  public string? Image { get; set; }
  public int CategoryId { get; set; }
  public Category? Category { get; set; } 
}

File: frontend/src/types/product.ts
import { Category } from './category';

export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  categoryName: string;
  image: string;
}

export interface SearchResultsProps {
  filteredProducts: Product[];
  searchTerm: string;
  onItemClick: () => void;
  showSearchResults: boolean;
}

export interface ProductCardProps {
  product: {
    id: number;
    image: string;
    categoryName: string;
    name: string;
    price: number;
  };
}
